src/serviceconnector-passwordless/azext_serviceconnector_passwordless/client_factory_util.py

```diff
@@ -0,0 +1,311 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core import _debug
+from azure.cli.core.auth.util import resource_to_scopes
+from azure.cli.core.extension import EXTENSIONS_MOD_PREFIX
+from azure.cli.core.profiles import ResourceType, CustomResourceType, get_api_version, get_sdk
+from azure.cli.core.profiles._shared import get_client_class, SDKProfile
+from azure.cli.core.util import get_az_user_agent, is_track2
+from knack.log import get_logger
+from knack.util import CLIError
+
+logger = get_logger(__name__)
+
+
+# pylint: disable=raise-missing-from, consider-using-f-string
+def _is_vendored_sdk_path(path_comps):
+    return len(path_comps) >= 5 and path_comps[4] == 'vendored_sdks'
+
+
+def resolve_client_arg_name(operation, kwargs):
+    if not isinstance(operation, str):
+        raise CLIError("operation should be type 'str'. Got '{}'".format(type(operation)))
```

Comment:
Please use other error types. `CLIError` is not recommended to use.

Reply:
Thanks for reminding. This file is just the copy of `client_factory.py` in core. We don't want to change the original implementation and just need to add extension name and version in user agent by adding 2 more lines.